parsers/ocaml.c

```diff
@@ -1509,38 +1632,57 @@ static void moduleSpecif (vString * const ident, ocaToken what)
 	case OcaKEYWORD_sig:
 		pushStrongContext (lastModule, ContextModule);
 		toDoNext = &globalScope;
+		needStrongPoping = FALSE;
 		break;
 
 	case Tok_PARL:	/* ( */
 		toDoNext = &contextualTillToken;
 		comeAfter = &globalScope;
 		waitedToken = Tok_PARR;
-		contextualTillToken (ident, what);
+		contextualTillToken (ident, what, whatNext);
+		break;
+
+	case Tok_Of:
+	case Tok_EQ:
 		break;
 
 	default:
 		vStringClear (lastModule);
 		toDoNext = &globalScope;
+		break;
 	}
 }
 
 /* parse :
  * module name = ...
  * then pass the token stream to moduleSpecif */
-static void moduleDecl (vString * const ident, ocaToken what)
+
+static void moduleDecl (vString * const ident, ocaToken what, ocaToken whatNext)
 {
 	switch (what)
 	{
+	case OcaKEYWORD_rec:
+		/* recursive modules are _weird_, but they happen */
 	case OcaKEYWORD_type:
-		/* just ignore it, name come after */
+		/* this is technically a special type, but whatever */
 		break;
```

Comment:
because a `module type` gives the type signature for the module, and could possibly used to build a more complete type annotation for the functions contained in the module itself, as types aren't explicitly annotated and usually inferred in OCaml, meaning ctags has _no_ idea what the types of anything are.